core/backend/src/annotations/TextBlockLayout.ts

```diff
@@ -255,36 +256,62 @@ export class TextStyleResolver {
     return style;
   }
 
-  /** Resolves the effective style for a [TextAnnotationLeader]($common). The TextAnnotationLeader should be a sibling of the provided TextBlock. */
-  public resolveTextAnnotationLeaderSettings(leader: TextAnnotationLeader): TextStyleSettings {
+  /**
+   * Resolves the effective text style settings for a given TextBlockComponent, applying block-level overrides.
+   * @param component The TextBlockComponent whose style settings are to be resolved.
+   * @returns The resolved TextStyleSettings for the component.
+   */
+  public resolveSettings(overrides: TextStyleSettingsProps, isLeader: boolean = false): TextStyleSettings {
     let settings = this.blockSettings;
 
-    if (leader.styleOverrides)
-      settings = settings.clone(leader.styleOverrides);
+    if (overrides)
+      settings = settings.clone(overrides);
 
-    return applyBlockSettings(settings, this.blockSettings, true);
+    return applyBlockSettings(settings, this.blockSettings, isLeader);
   }
 
-  /** Resolves the effective style for a [Paragraph]($common). Paragraph should be child of provided TextBlock. */
-  public resolveParagraphSettings(paragraph: Paragraph): TextStyleSettings {
-    return applyBlockSettings(this.resolveParagraphSettingsImpl(paragraph), this.blockSettings);
+  /**
+   * Computes the indentation for a TextBlockComponent based on its style and nesting depth.
+   * @param component The TextBlockComponent for which to compute indentation.
+   * @param depth The nesting depth of the component.
+   * @returns The computed indentation value.
+   */
+  public resolveIndentation(component: TextBlockComponent, depth: number): number {
+    const overrides = this.resolveSettings(component.styleOverrides);
+    const indentation = overrides.indentation;
+    const tabInterval = overrides.tabInterval;
+    return indentation + tabInterval * depth;
   }
 
-  /** Resolves the effective style for a [Run]($common). Run should be child of provided Paragraph and TextBlock. */
-  public resolveRunSettings(paragraph: Paragraph, run: Run): TextStyleSettings {
-    let settings = this.resolveParagraphSettingsImpl(paragraph);
+  /**
+   * Resolves and applies style overrides to a TextBlock, ensuring all child components are updated.
+   * @param component The TextBlock to resolve and mend styles for.
+   * @returns A cloned TextBlock with resolved and mended styles.
+   */
+  public resolveAndMendStyle(component: TextBlock): TextBlock {
```

Comment:
Get rid of all this "clone and mend settings" stuff as directed in your TODO in `populateComponent`.